backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,334 @@
+
+			"skip_cert_verification": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "(Optional) Whether to skip TLS verification. Defaults to false.",
+			},
+
+			"local_cert_ca_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "CA to use when the rest api is using a self signed certificate.",
+			},
+
+			"mutual_tls_authentication": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Use mutual tls authentication.local_cert_file, local_key_file, local_ca_file needs to be set. Defaults to false.",
+			},
+
+			"local_cert_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Cert file to be used for mutual tls authentication.",
+			},
+
+			"local_key_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Key file to be used for mutual tls authentication.",
+			},
+
+			"local_ca_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "CA to be used for mutual tls authentication.",
+			},
+		},
+	}
+
+	result := &Backend{Backend: s}
+	result.Backend.ConfigureFunc = result.configure
+	return result
+}
+
+// Backend Structure
+type Backend struct {
+	*schema.Backend
+	data   *schema.ResourceData
+	client *http.Client
+
+	// The fields below are set from configure
+	address         string
+	updateMethod    string
+	lockAddress     string
+	lockMethod      string
+	unlockMethod    string
+	unlockAddress   string
+	skipTLS         bool
+	localCertCAFile string
+	localCertFile   string
+	localKeyFile    string
+	localCAFile     string
+	mutualTLS       bool
+	username        string
+	password        string
+}
+
+func (b *Backend) configure(ctx context.Context) error {
+	if b.client != nil {
+		return nil
+	}
+	// Grab the resource data
+	data := schema.FromContextBackendConfig(ctx)
+
+	// Store the URL address
+	addressURL := data.Get("address").(string)
+	// Validate URL address
+	isValidURL(addressURL)
+	b.address = addressURL
+
+	b.updateMethod = data.Get("update_method").(string)
+
+	if v, ok := data.GetOk("lock_address"); ok {
+		lockAddress := v.(string)
+		// Validate lockAddress
+		isValidURL(lockAddress)
+		b.lockAddress = lockAddress
+	} else {
+		// If lockAddress is null, use the http rest api address
+		b.lockAddress = b.address
+	}
+
+	b.lockMethod = data.Get("lock_method").(string)
+
+	if v, ok := data.GetOk("unlock_address"); ok {
+		unlockAddress := v.(string)
+		// Validate unlockAddress
+		isValidURL(unlockAddress)
+		b.unlockAddress = unlockAddress
+	} else {
+		// If unlockAddress is null, use the http rest api address
+		b.unlockAddress = b.address
+	}
+
+	b.unlockMethod = data.Get("unlock_method").(string)
+
+	if v, ok := data.GetOk("username"); ok {
+		b.username = v.(string)
+	}
+
+	if v, ok := data.GetOk("password"); ok {
+		b.password = v.(string)
+	}
+
+	client := &http.Client{}
```

Comment:
In all of the following logic, you override this client. Essentially all the following logic is doing is configuring the `client.Transport` attribute.  I think you should setup the base client attributes here instead of an empty struct. You can put your common stuff here, like so:

```
client := &http.Client{
    Timeout: time.Second * 10,
}
```

You can also define default HTTP transport options as part of this scope:

```
transport := &http.Transport{
    // Default transport config here
}
```

Instead of overriding the entire client, you would just update the `transport` by setting the correct attributes.

Then at the end of this function, just set `client.Transport = transport`

Reply:
I override this client only in certain conditions. If not, it will use this one. Good catch, forgot to add the timeout here. I know it's a big think with this timeout.
https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779